src/NATS.Client.Core/NatsMsg.cs

```diff
@@ -207,6 +321,16 @@ internal static NatsMsg<T> Build(
         INatsDeserialize<T> serializer)
     {
         NatsHeaders? headers = null;
+        var flags = NatsMsgFlags.None;
+
+        if (payloadBuffer.Length == 0)
+        {
+            flags |= NatsMsgFlags.Empty;
+            if (NatsSubBase.IsHeader503(headersBuffer))
```

Comment:
Should this be unnested?

Reply:
it's pretty specific [where server sets the 503 header for no responders](https://github.com/nats-io/nats-server/blob/290f4095a21392175a5a158167efd31cfc81f595/server/client.go#L4081) and [Go client seems to check for data being empty](https://github.com/nats-io/nats.go/blob/94fa0cb99411c202a4c9c2ae84f444a10f2e5ad8/nats.go#L4074) too. tbh I think no-responders is the only usage of 503 so i suppose it would be ok to move it out idk.

Reply:
Oh nvm I guess I didn't realize they noresponders only came with empty 😅